tests/js/client/shell/api/cursor.js

```diff
@@ -1212,9 +1212,268 @@ function dealing_with_cursorsSuite_fetching_floating_point_valuesSuite () {
   };
 }
 
+////////////////////////////////////////////////////////////////////////////////;
+// cursor with retriable request for the last batch;
+////////////////////////////////////////////////////////////////////////////////;
+
+function dealing_with_cursorsSuite_retriable_request_last_batch() {
+  let cn = "users";
+  return {
+    setUpAll: function () {
+      db._drop(cn);
+      let c = db._create(cn);
+
+      let docs = [];
+      for (let i = 0; i < 2001; i++) {
+        docs.push({"_key": `test${i}`});
+      }
+      c.insert(docs);
+    },
+
+    tearDownAll: function () {
+      internal.debugClearFailAt();
+      db._drop(cn);
+    },
+
+    test_cursor_non_stream_request_retriable: function () {
+      let cmd = api;
+      let body = {"query": `FOR u IN ${cn} RETURN u`, "options": {"stream": false, "retriable": true}};
+      let doc = arango.POST_RAW(cmd, body);
+
+      assertEqual(doc.code, 201);
+      assertEqual(doc.headers['content-type'], contentType);
+      assertFalse(doc.parsedBody['error']);
+      assertEqual(doc.parsedBody['code'], 201);
+      assertEqual(typeof doc.parsedBody['id'], "string");
+      assertMatch(reId, doc.parsedBody['id']);
+      assertTrue(doc.parsedBody['hasMore']);
+      assertEqual(doc.parsedBody['result'].length, 1000);
+      assertFalse(doc.parsedBody['cached']);
+
+      let cursorId = doc.parsedBody['id'];
+      assertEqual(doc.parsedBody['nextBatchId'], 2);
+      let nextBatchId = doc.parsedBody['nextBatchId'];
```

Comment:
Is `nextBatchId` an integer or a string? I think we should also check its type here. 
```suggestion
      let nextBatchId = doc.parsedBody['nextBatchId'];
      assertEqual(nextBatchId, "2");
      
```